torchmetrics/text/squad.py

```diff
@@ -0,0 +1,198 @@
+        self.add_state(name="total", default=[], dist_reduce_fx=None)
+
+    def update(
+        self,
+        preds: PREDS_TYPE,
+        targets: TARGETS_TYPE,
+    ) -> None:  # type: ignore
+        """Compute F1 Score and Exact Match for a collection of predictions and references.
+
+        Args:
+            preds:
+                A Dictionary or List of Dictionary-s that map `id` and `prediction_text` to the respective values.
+
+                Example prediction:
+
+                .. code-block:: python
+
+                    {"prediction_text": "TorchMetrics is awesome", "id": "123"}
+
+            targets:
+                A Dictioinary or List of Dictionary-s that contain the `answers` and `id` in the SQuAD Format.
+
+                Example target:
+
+                .. code-block:: python
+
+                    {
+                        'answers': [{'answer_start': [1], 'text': ['This is a test answer']}],
+                        'id': '1',
+                    }
+
+                Reference SQuAD Format:
+
+                .. code-block:: python
+
+                    {
+                        'answers': {'answer_start': [1], 'text': ['This is a test text']},
+                        'context': 'This is a test context.',
+                        'id': '1',
+                        'question': 'Is this a test?',
+                        'title': 'train test'
+                    }
+
+        Raises:
+            KeyError:
+                If the required keys are missing in either predictions or targets.
+        """
+        if isinstance(preds, Dict):
+            preds = [preds]
+
+        if isinstance(targets, Dict):
+            targets = [targets]
+
+        for pred in preds:
+            keys = pred.keys()
+            if "prediction_text" not in keys or "id" not in keys:
+                raise KeyError(
+                    "Expected keys in a single prediction are 'prediction_text' and 'id'."
+                    "Please make sure that 'prediction_text' maps to the answer string and 'id' maps to the key string."
+                )
+
+        for target in targets:
+            keys = target.keys()
+            if "answers" not in keys or "id" not in keys:
+                raise KeyError(
+                    "Expected keys in a single target are 'answers' and 'id'."
+                    "Please make sure that 'answers' maps to a `SQuAD` format dictionary and 'id' maps to the key"
+                    "string.\n"
+                    "SQuAD Format: "
+                    "{"
+                    "    'answers': {"
+                    "        'answer_start': [1],"
+                    "        'text': ['This is a test text']"
+                    "    },"
+                    "    'context': 'This is a test context.',"
+                    "    'id': '1',"
+                    "    'question': 'Is this a test?',"
+                    "    'title': 'train test'"
+                    "}"
```

Comment:
duplicate in text, please define some variable and just reuse
```suggestion
                    f"{squard_format}"
```